### PR TITLE
Use envoy with LoadBalancerClass to expose Tinkerbell services

### PR DESCRIPTION
**Pull request by abhinavmpandey08**
*Issue #, if available:*
https://github.com/aws/eks-anywhere/issues/2745, https://github.com/aws/eks-anywhere/issues/2772

*Description of changes:*
This PR does a few things:
- Exposes `hegel` and `tink-server` externally using envoy and a single load balancer service
- Uses LoadBalancerClass for the service to ensure only kube-vip handles the service
- Re-organizes the templates so they are easier to maintain and follow

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by abhay-krishna** [comment]
nit: why do we need the `<tink service>-` prefix for the template files if the folder name itself is `<tink service>`.

**Comment by abhinavmpandey08** [comment]
> nit: why do we need the `<tink service>-` prefix for the template files if the folder name itself is `<tink service>`?

Removed the service specific prefix from the templates

File: projects/tinkerbell/tinkerbell-chart/chart/values.yaml [hunk 1]
```diff
@@ -26,12 +26,13 @@ boots:
...
-  image: quay.io/tinkerbell/hegel:latest
+  image: public.ecr.aws/eks-anywhere/tinkerbell/hegel:7b286fdc8e8fa91a6e9a179a5494b6ee29fce17b-eks-a-12
   imagePullPolicy: IfNotPresent
+  replicas: 1
   port:
-    hostPortEnabled: true
+    hostPortEnabled: false
     hostPort: 50061
-  args: []
+  args: ["--grpc-use-tls=false"]
```
Comment (ptrivedi) [inline projects/tinkerbell/tinkerbell-chart/chart/values.yaml:35]: why was this needed?

Comment (abhinavmpandey08) [reply projects/tinkerbell/tinkerbell-chart/chart/values.yaml:35]: We override this value from the CLI so it shouldn't matter anyways
https://github.com/aws/eks-anywhere/blob/main/pkg/providers/tinkerbell/stack/stack.go#L154
